### DISCUSSION TITLE
Thinner lines into 1px pencil strokes

### DISCUSSION THREAD

**Discussion by mrx23dot** [ImageMagick/ImageMagick#5169] [Help] [answered]
Any way I could convert this into 1px line, like you would redraw it by hand in inkscape.
I would also want to avoid double lines (not outline).

![2022-05-24 15_07_43-guiout png](https://user-images.githubusercontent.com/4163396/170055810-c77343ba-29e9-444a-afaa-ce86dbd64b49.jpg)

Goal:
![Untitled](https://user-images.githubusercontent.com/4163396/170058428-c18b45d5-d260-4bc5-8c6b-00690dcc9af6.png)

E.g. clip-studio hash raster line thickening/thinning function:
https://www.clip-studio.com/site/gd_en/csp/userguide/csp_userguide/500_menu/500_menu_filter_line_senhaba.htm

**Comment by snibgo** [comment, marked as answer]
Windows BAT script:
```
%IMG7%magick ^
  shoulder.jpg ^
  -threshold 50%% ^
  -negate ^
  -morphology Erode Diamond ^
  -morphology Thinning:-1 Skeleton ^
  x1.png

call %PICTBAT%pruneStubs x1.png x2.png 10

%IMG7%magick ^
  x2.png ^
  -negate ^
  x3.png
```
See http://imagemagick.org/Usage/morphology/ and [Pruning skeleton stubs](http://im.snibgo.com/prunestubs.htm).
![x3](https://user-images.githubusercontent.com/33812368/170072615-e04f6bf2-94bc-4c1c-ad8a-a6cb6360f7f3.png)

**Reply by mrx23dot** [reply]
wow, looks close enough, thank you!
